### ISSUE TITLE
Check file magic before opening png with Cairo

### ISSUE THREAD

**Issue by lcarlsen** [andymeneely/squib#250]
Inserting a png fails with an unexpected error message when the file isn't really a png.  Error is `vendor/bundle/ruby/2.3.0/bundler/gems/squib-01e0d77f82d8/lib/squib/graphics/image.rb:9:in` `from_png': out of memory (NoMemoryError)`.  Expected error would be something like 'File <filename> is not a png.'

I encountered this issue on Ubuntu 16.04.  Version of libcairo2 is 1.14.6-1.

**Comment by andymeneely** [comment]
Yeah we can catch this and do a better message. Unfortunately NoMemoryError at the Cairo level can mean a whole lot of different things. So maybe it'll be:

"Cairo couldn't open this PNG file - are you sure it's a well-formed png? Original error: ..."

Thoughts?

**Comment by lcarlsen** [comment]
After looking through Cairo source, I think it's better to check the magic bits after a cache miss in image.rb, before invoking Cairo's from_png.  I'll file a separate bug against Cairo if I can make a small test case in C, but that will take longer.  No need to use filemagic gem just for this case; just a couple lines of Ruby to read 8 bytes and compare.

**Comment by andymeneely** [comment]
Ah, that's easy enough to do. Yeah I'll work that in.

**Comment by andymeneely** [comment]
Just about done. This'll be in the next 0.15.0 release. Thanks! 

This is also related to #218.
